src/traffic_server/traffic_server.cc

```diff
@@ -2015,42 +2016,37 @@ REGRESSION_TEST(Hdrs)(RegressionTest *t, int atype, int *pstatus)
 }
 #endif
 
-static void *
-mgmt_restart_shutdown_callback(void *, char *, int /* data_len ATS_UNUSED */)
+static void mgmt_restart_shutdown_callback(ts::MemSpan)
 {
   sync_cache_dir_on_shutdown();
-  return nullptr;
 }
 
-static void *
-mgmt_drain_callback(void *, char *arg, int len)
+static void
+mgmt_drain_callback(ts::MemSpan span)
 {
-  ink_assert(len > 1 && (arg[0] == '0' || arg[0] == '1'));
+  char *arg = static_cast<char *>(span.data());
+  ink_assert(span.size() > 1 && (arg[0] == '0' || arg[0] == '1'));
   RecSetRecordInt("proxy.node.config.draining", arg[0] == '1', REC_SOURCE_DEFAULT);
-  return nullptr;
 }
 
-static void *
-mgmt_storage_device_cmd_callback(void *data, char *arg, int len)
+static void
+mgmt_storage_device_cmd_callback(int cmd, std::string_view const &arg)
 {
   // data is the device name to control
-  CacheDisk *d = cacheProcessor.find_by_path(arg, len);
-  // Actual command is in @a data.
-  intptr_t cmd = reinterpret_cast<intptr_t>(data);
+  CacheDisk *d = cacheProcessor.find_by_path(arg.data(), int(arg.size()));
 
   if (d) {
     switch (cmd) {
     case MGMT_EVENT_STORAGE_DEVICE_CMD_OFFLINE:
-      Debug("server", "Marking %.*s offline", len, arg);
+      Debug("server", "Marking %.*s offline", int(arg.size()), arg.data());
```

Comment:
Using a non-Leif-legal constructor-style cast here.

Reply:
I'm going to blame you.